frontend/awx/resources/inventories/InventoryForm.cy.tsx

```diff
@@ -0,0 +1,344 @@
+import { CyHttpMessages } from 'cypress/types/net-stubbing';
+import { AwxItemsResponse } from '../../common/AwxItemsResponse';
+import { InstanceGroup } from '../../interfaces/InstanceGroup';
+import { Inventory } from '../../interfaces/Inventory';
+import { Label } from '../../interfaces/Label';
+import { Organization } from '../../interfaces/Organization';
+import { CreateInventory, EditInventory, InventoryCreate } from './InventoryForm';
+
+export type regularPayload = {
+  kind: string;
+  name: string;
+  description: string;
+  labels: Array<{ name: string }>;
+  variables: string;
+  prevent_instance_group_fallback: boolean;
+  organization: number;
+};
+
+export type smartPayload = {
+  kind: string;
+  name: string;
+  description: string;
+  variables: string;
+  organization: number;
+  host_filter: string;
+};
+
+describe('Create Edit Inventory Form', () => {
+  const regularPayload = {
+    kind: '',
+    name: 'test',
+    description: 'test description',
+    labels: [{ name: 'test label' }],
+    variables: 'hello:world',
+    prevent_instance_group_fallback: false,
+    organization: 1,
+  };
+
+  const smartPayload = {
+    kind: 'smart',
+    name: 'smart test',
+    description: 'smart test description',
+    variables: 'hello:world',
+    organization: 1,
+    host_filter: 'name__icontains=local',
+  };
+
+  const kinds: ('' | 'smart' | 'constructed')[] = ['', 'smart'];
+
+  describe('Create Inventory', () => {
+    beforeEach(() => {
+      cy.intercept(
+        { method: 'GET', url: '/api/v2/organizations/*' },
+        { fixture: 'organizations.json' }
+      );
+      cy.intercept(
+        { method: 'GET', url: '/api/v2/instance_groups/*' },
+        { fixture: 'instance_groups.json' }
+      );
+      cy.intercept({ method: 'GET', url: '/api/v2/labels/*' }, { fixture: 'labels.json' });
+    });
+    kinds.forEach((kind) => {
+      const path = '/inventories/:inventory_type/create';
+
+      const initialEntries =
+        kind === '' ? [`/inventories/inventory/create`] : [`/inventories/smart_inventory/create`];
+
+      const payload: regularPayload | smartPayload = kind === '' ? regularPayload : smartPayload;
+
+      it(`Validate required fields on create (${kind === '' ? 'regular' : kind})`, () => {
```

Comment:
${kind === '' ? 'regular' : kind}

This seems to be repeated a lot, what about create it as constant once and then reuse it in every it description?